sandpiper/conversion/unit_conversion.py

```diff
@@ -47,7 +47,7 @@ def __getitem__(self, item):
 })
 
 imperial_height_pattern = re.compile(
-    r'^(?=.)(?:(?P<foot>[\d]+)\')?(?:(?(foot) |)(?P<inch>[\d.]+)\")?$')
```

Comment:
Most of this reduction was me not knowing what it was for and not seeing how it affected the test cases. If it's required please add it back =P

Reply:
So at the beginning, there's a positive lookahead to ensure that there are SOME characters in here to try matching. Since the whole pattern is wrapped a zero-or-one quantifier, I think it could match a zero-length string: `"{}"`.

After that, the whole pattern is wrapped in a non-capturing group just because it's good practice to not use unnecessary capturing groups. It really doesn't matter here because it's not performance-critical and we use named capturing groups instead of numbered, but I'll just revert it since I'm gonna update the pattern anyway.

I think I'm going to re-add the conditional you removed just so it doesn't match leading spaces if only the inches are specified: `   5"`. It'll ensure spaces before inches is only valid if feet is specified. If we want space stripping, we could add it somewhere else explicitly.

I also *attempted* support for decimal inch values, but I somehow missed the fact that `1.2.3.4` is also valid with the `[\d.]+` pattern... 😆 There's also the potential for adding decimal feet values, but I'm gonna skip that for now.

Reply:
That's probably because "." matches all in regex =P

Reply:
Also the regex I used doesn't match empty lines in my tests but not sure if that translates to the chat messages

Reply:
> 
> 
> That's probably because "." matches all in regex =P

Inside brackets though, periods are literal! `[.]` only matches `"."`. Weird stuff like `[[]` is also valid.

Reply:
> 
> 
> Also the regex I used doesn't match empty lines in my tests but not sure if that translates to the chat messages

I also tried empty lines on regex101, and it doesn't match those, but in Python it is matching empty strings.

Reply:
> > That's probably because "." matches all in regex =P
> 
> Inside brackets though, periods are literal! `[.]` only matches `"."`. Weird stuff like `[[]` is also valid.

Good to know =P

> > Also the regex I used doesn't match empty lines in my tests but not sure if that translates to the chat messages
> 
> I also tried empty lines on regex101, and it doesn't match those, but in Python it is matching empty strings.

Dangit Python >.<

Anyway -- as long as the spaces are still optional afterwards you can revert literally everything =P